### ISSUE TITLE
Ensure that we support x[[sym]]

### ISSUE THREAD

**Issue by hadley** [r-lib/vctrs#575]
As base R does:

``` r
list(a = 1)[[quote(a)]]
#> [1] 1
c(a = 1)[[quote(a)]]
#> [1] 1
data.frame(a = 1)[[quote(a)]]
#> [1] 1
```

My intuition is that will tend to help with certain tidy evaluation tasks, and even if not, it's a relatively obscure feature so there shouldn't be downsides associated with implementing it.

**Comment by krlmlr** [comment]
This seems to break the interaction package with dev tibble: https://github.com/tidyverse/tibble/blob/master/revdep/problems.md#interactions.

**Comment by krlmlr** [comment]
Appears to work even in vctrs 0.2.3:

``` r
vctrs::vec_as_location(quote(c), 26, letters)
#> [1] 3
vctrs::vec_as_location2(quote(c), 26, letters)
#> [1] 3
```

<sup>Created on 2020-02-27 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by lionel-** [comment]
Not with `vec_slice()`, but this will be fixed on master soon once #868 is merged.
